app/controllers/connection/git_hub_controller.rb

```diff
@@ -4,7 +4,13 @@ class Connection::GitHubController < ApplicationController
   skip_before_action :require_active_user_login, raise: false
 
   def destroy
-    current_user.update(github_id: nil)
-    redirect_to root_path, notice: 'GitHubとの連携を解除しました。'
+    user = User.find(params[:user_id])
+    if !admin_login? && user != current_user
+      redirect_to root_path, alert: '管理者としてログインしてください'
+      return
+    end
+
+    user.update(github_id: nil)
+    redirect_to user_path(user), notice: 'GitHubとの連携を解除しました。'
```

Comment:
ガード節以外ではreturnを使わず下記のような感じの方が読みやすいのではないかと思いました。

```suggestion
    if !admin_login? && user != current_user
      redirect_to root_path, alert: '管理者としてログインしてください'
    else
      user.update(github_id: nil)
      redirect_to user_path(user), notice: 'GitHubとの連携を解除しました。'
    end
```

Reply:
コメントありがとうございます。
ご提案いただいたコードの方が読みやすいと感じましたので修正しました。